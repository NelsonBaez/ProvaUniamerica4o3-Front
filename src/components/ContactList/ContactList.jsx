import React, { useEffect, useState } from 'react';
import { Outlet, useParams } from 'react-router';
import api from '../../api/api';
import ContactLink from '../ContactLink/ContactLink';


export default function ContactList(){
  const [contacts, setContacts] = useState([]);
  const history = useParams();
  console.log(history);

  let newContact = {
    id: 'new',
    name: 'Cadastrar novo Contato'
  }

  function getContacts(){
    api.get('/contacts')
      .then( (response) => setContacts(response.data))
      .catch((err) => {
        console.error(err);
      })
  }
  useEffect(() => {
    getContacts();
  }, [history]);
  
  return (
    <div>
      <div>
          <ContactLink contact={newContact}  />
        {
          contacts.map((contact) => {
            return <ContactLink contact={contact} />
          })
        }
      </div>
        <Outlet />
    </div>
  )
};