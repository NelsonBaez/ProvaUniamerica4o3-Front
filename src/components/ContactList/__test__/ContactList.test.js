import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import ContactList from '../ContactList';
import axios from 'axios';
import nock from 'nock';



afterEach(function() {
  nock.cleanAll();
});

// const MockMyComponent = () => {
//   React.useEffect(() => {
//     console.log('using an effect');
//   });
//   return (<div>Hello World</div>);
// };
// jest.mock('./MyComponent', () => MockMyComponent);

// test("renders contacts", () => {
  
//   render(
//     <MemoryRouter><ContactList /></MemoryRouter>
//   )
//   const element = screen.getByText('jorge');
//   expect(element).toBeInTheDocument();
//   expect(element).toHaveTextContent('jorge');
// })

test('matches snapshot', () => {
  const tree = renderer.create(
    
    <MemoryRouter><ContactList /></MemoryRouter>
  
  ).toJSON();
  expect(tree).toMatchSnapshot();
  
});