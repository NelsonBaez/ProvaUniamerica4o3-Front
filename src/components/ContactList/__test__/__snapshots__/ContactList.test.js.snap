// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div>
      <a
        className="blue"
        data-testid="link-new"
        href="/contatos/new"
        onClick={[Function]}
      >
        Cadastrar novo Contato
      </a>
    </div>
  </div>
</div>
`;
